kroxylicious/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyInitializer.java

```diff
@@ -58,16 +56,51 @@ public void initChannel(SocketChannel ch) {
 
         ChannelPipeline pipeline = ch.pipeline();
 
-        sslContext.ifPresent(s -> {
+        int targetPort = ch.localAddress().getPort();
```

Comment:
This is the guts of the change.. uses either SNI/target port or target port to identify which virtualcluster is being connected to.